src/model/engines/engine_base.nit

```diff
@@ -0,0 +1,59 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Base for code runners of a kind
+module engine_base
+
+import submissions
+
+# Any class capable of running some code in a defined language
+class Engine
+	# Which language is supported by the engine?
+	fun language: String is abstract
+
+	# Compile and run the tests for `program`
+	fun run(program: Program) do
+		program.status = "pending"
+		var runnable = compile(program)
+		if not runnable then
+			program.test_errors = 1
+			# Dirty trick to force infinity
+			program.time_score = inf
```

Comment:
Another solution for the stars may be to let the time be calculated normally, but the star is only awarded when no error has been detected (neither during compilation nor testing)